samcli/commands/sync/core/command.py

```diff
@@ -0,0 +1,142 @@
+"""
+Sync Command Class.
+"""
+from typing import List
+
+from click import Command, Context, Parameter, style
+
+from samcli.cli.row_modifiers import RowDefinition, ShowcaseRowModifier
+from samcli.commands.sync.core.formatters import SyncCommandHelpTextFormatter
+from samcli.commands.sync.core.options import OPTIONS_INFO
+
+
+class SyncCommand(Command):
+    class CustomFormatterContext(Context):
+        formatter_class = SyncCommandHelpTextFormatter
+
+    context_class = CustomFormatterContext
+
+    def __init__(self, description, requires_credentials=False, *args, **kwargs):
+        self.description = description
+        self.requires_credentials = requires_credentials
+        self.description_addendum = (
+            style("\n  This command requires access to AWS credentials.", bold=True)
+            if self.requires_credentials
+            else style("\n  This command does not require access to AWS credentials.", bold=True)
+        )
+        super().__init__(*args, **kwargs)
+
+    def format_description(self, formatter: SyncCommandHelpTextFormatter):
+        with formatter.indented_section(name="Description", extra_indents=1):
+            formatter.write_rd(
+                [
+                    RowDefinition(
+                        text="",
+                        name=self.description + self.description_addendum,
+                    ),
+                ],
+            )
+
+    @staticmethod
+    def format_examples(ctx: Context, formatter: SyncCommandHelpTextFormatter):
+        with formatter.indented_section(name="Examples", extra_indents=1):
+            formatter.write_rd(
+                [
+                    RowDefinition(
+                        name="",
+                        text="\n",
+                    ),
+                    RowDefinition(
+                        text="",
+                        name=style(f"${ctx.command_path} " f"--code --watch --stack-name {{stack}}"),
+                        extra_row_modifiers=[ShowcaseRowModifier()],
+                    ),
+                    RowDefinition(
+                        text="",
+                        name=style(
+                            f"${ctx.command_path} "
+                            f"--code --stack-name {{stack}} --resource-id {{ChildStack}}/{{ResourceId}} "
```

Comment:
Can we also add an example, which will basically be `sam sync --watch --stack-name {stack}`?

Reply:
Definitely!